dart/lib/src/protocol/sentry_exception.dart

```diff
@@ -86,10 +88,25 @@ class SentryException {
         type: type ?? this.type,
         value: value ?? this.value,
         module: module ?? this.module,
-        stackTrace: stackTrace ?? this.stackTrace,
-        mechanism: mechanism ?? this.mechanism,
+        stackTrace: stackTrace ?? this.stackTrace?.copyWith(),
+        mechanism: mechanism ?? this.mechanism?.copyWith(),
```

Comment:
do we need the copyWith?

Reply:
For the tests, yeah. There we use copy with and without this, we'd just share the same instance of stackTrace/mecahnism between multiple tests, now that it's immutable.

Reply:
Let's leave it for now and handle this some other time